Sources/Atoms/Core/TopologicalSort.swift

```diff
@@ -0,0 +1,72 @@
+internal enum Vertex: Hashable {
+    case atom(key: AtomKey)
+    case subscriber(key: SubscriberKey)
+}
+
+internal struct Edge: Hashable {
+    let from: AtomKey
+    let to: Vertex
+}
+
+/// DFS topological sorting.
+@MainActor
+internal func topologicalSort(key: AtomKey, store: AtomStore) -> (
```

Comment:
Global main actor func just to access main actor isolated `store.graph`, why not make it extension of `AtomStore` which already is on mainactor.

[nit]
Also, do you think `topologicalSort` is a proper name given it returns edges? I can suggest to make it something more fitting the caller site → `update(atom...)` , e.g. edgesToUpdate() etc, and then hide the technique with private method like topologicallySorted() -> edges, redundants..

Reply:
> Global main actor func just to access main actor isolated store.graph, why not make it extension of AtomStore which already is on mainactor.

Sounds good to me.

> do you think topologicalSort is a proper name given it returns edges?

I believe how it is sorted is important for the caller as the additional algorithm on the StoreContext doesn't work except with topological sorting.
Also, any other naming wouldn't justify it returning the pair of edges and redundant.
More descriptive naming like `topologicallySortedEdgesWithRedundantPath` would make it clear but it sounds too much.

Reply:
I'm already on a bunch of refactoring in another [branch](https://github.com/ra1028/swiftui-atom-properties/pull/127), so will apply it there.